### PR TITLE
improve documentation for fields package

### PR DESCRIPTION
**Pull request by gigitux**
## What?

This pull request includes documentation updates for various actions and fields in the `packages/fields` package. Also, remove unused native files.

I think that there is still some space for improvements (especially on action names) but we can revisit later.

### CODE REVIEW

File: packages/fields/src/fields/template/index.ts [hunk 1]
```diff
@@ -14,9 +14,11 @@ const templateField: Field< BasePost > = {
 	id: 'template',
 	type: 'text',
 	label: __( 'Template' ),
-	getValue: ( { item } ) => item.template,
```
Comment (gigitux) [inline packages/fields/src/fields/template/index.ts:17]: This is not necessary. I removed it. https://github.com/WordPress/gutenberg/pull/66591#discussion_r1869465438

File: packages/fields/src/actions/duplicate-post.native.tsx [hunk 2]
```diff
@@ -1,3 +0,0 @@
-const duplicatePost = undefined;
-
-export default duplicatePost;
```
Comment (oandregal) [inline packages/fields/src/actions/duplicate-post.native.tsx:3]: Why do we delete this file? Is it not used by mobile (I see it's named `*.native.*`)?

File: packages/fields/src/actions/duplicate-post.tsx [hunk 3]
```diff
@@ -37,8 +37,8 @@ const duplicatePost: Action< BasePost > = {
 		const [ item, setItem ] = useState< BasePost >( {
 			...items[ 0 ],
 			title: sprintf(
-				/* translators: %s: Existing template title */
```
Comment (oandregal) [inline packages/fields/src/actions/duplicate-post.tsx:40]: Nice catch, thanks! We actually [hide](https://github.com/WordPress/gutenberg/blame/be60bebeff7f6f699fd33d1c35840d9b549cbaa7/packages/editor/src/dataviews/store/private-actions.ts#L137) this action for templates.

File: packages/fields/src/actions/export-pattern.native.tsx [hunk 4]
```diff
@@ -1,3 +0,0 @@
-const exportPattern = undefined;
-
-export default exportPattern;
```
Comment (oandregal) [inline packages/fields/src/actions/export-pattern.native.tsx:3]: Same as above: isn't this used by mobile?

**Comment by oandregal** [comment]
Thanks for taking the time :)

I'd approve but I'd like to double-check why we're removing the `*.native.*` files used by mobile.

**Comment by gigitux** [comment]
> Thanks for taking the time :)
> 
> I'd approve but I'd like to double-check why we're removing the `*.native.*` files used by mobile.

These two actions exports`undefined`. Also, if you search for `exportPatternNative`, `duplicatePostNative` and `reorderPageNative`, I can't find any reference:

image

image


image

Also, mobile build doesn't fail, so I think that these changes are fine.

**Comment by oandregal** [comment]
> Also, mobile build doesn't fail, so I think that these changes are fine.

They actually don't run :) There's an issue with the runner mobile folks are looking at https://github.com/WordPress/gutenberg/pull/67586

File: packages/fields/src/actions/index.ts [hunk 5]
```diff
@@ -1,13 +1,10 @@
 export { default as viewPost } from './view-post';
 export { default as reorderPage } from './reorder-page';
-export { default as reorderPageNative } from './reorder-page.native';
```
Comment (oandregal) [inline packages/fields/src/actions/index.ts:3]: @dcalhoun sorry to ping you so many times lately :)

Is it ok to remove the native actions here? They export `undefined` ([reorder](https://github.com/WordPress/gutenberg/pull/67580/files#diff-92034ee995c7df98ec7822103430854fdbfba95884a33c2207b89ccf8d957b2d)), [export pattern](https://github.com/WordPress/gutenberg/pull/67580/files#diff-adb87917ac23eb2f91ab752cbf792b63ef7afff13e909812384b193c57727529), [duplicate post](https://github.com/WordPress/gutenberg/pull/67580/files#diff-bee647c4d0b8f8af18507649b5372b0b6427660454a84d055ae80b0dc8be4579), so we wonder why are they necessary?

Comment (youknowriad) [reply packages/fields/src/actions/index.ts:3]: Mostly to make unit tests pass. If nothing breaks, it's ok to remove IMO

Comment (dcalhoun) [reply packages/fields/src/actions/index.ts:3]: I'll concede to @youknowriad's knowledge on this one. It appears the addition occurred in https://github.com/WordPress/gutenberg/pull/64199/commits/94d1d2b7f1095fc5692f0f073160badfcbd0130f to repair unit test failures. I agree it is safe to remove if unit tests continue passing.

Comment (oandregal) [reply packages/fields/src/actions/index.ts:3]: Thanks for the confirmation to both of you 🙇

**Comment by oandregal** [comment]
Given the native tests are blocked, I ran them locally (`npm run test:native`) and they pass:

Screenshot 2024-12-05 at 10 40 59

**Review by oandregal — APPROVED** [review]

**Comment by gigitux** [comment]
Thanks for the feedback and review folks! 🙇

**Pull request merged**